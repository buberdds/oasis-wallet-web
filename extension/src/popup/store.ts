import { Store, applyMiddleware } from 'webext-redux'
import createSagaMiddleware from 'redux-saga'

export default function configureAppStore() {
  const store = new Store()
  const sagaMiddleware = createSagaMiddleware({})
  const middlewares = [sagaMiddleware]
  const storeWithMiddleware = applyMiddleware(store, ...middlewares)
  return storeWithMiddleware
}
