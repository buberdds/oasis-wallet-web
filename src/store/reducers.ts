/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers } from '@reduxjs/toolkit'
import createWalletReducer from 'app/pages/CreateWalletPage/slice'
import openWalletReducer from 'app/pages/OpenWalletPage/slice'
import accountReducer from 'app/state/account'
import fatalErrorReducer from 'app/state/fatalerror'
import ledgerReducer from 'app/state/ledger'
import networkReducer from 'app/state/network'
import stakingReducer from 'app/state/staking'
import transactionReducer from 'app/state/transaction'
import walletReducer from 'app/state/wallet'
import themeReducer from 'styles/theme/slice'

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export function createReducer(injectedReducers = {}) {
  // Initially we don't have any injectedReducers, so returning identity function to avoid the error
  const rootReducer = combineReducers({
    ...injectedReducers,
    account: accountReducer,
    createWallet: createWalletReducer,
    fatalError: fatalErrorReducer,
    ledger: ledgerReducer,
    network: networkReducer,
    openWallet: openWalletReducer,
    staking: stakingReducer,
    theme: themeReducer,
    transaction: transactionReducer,
    wallet: walletReducer,
  })

  return rootReducer
}
