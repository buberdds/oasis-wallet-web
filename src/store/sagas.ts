import { all } from 'typed-redux-saga'

import { accountSaga } from 'app/state/account/saga'
import { networkSaga } from 'app/state/network/saga'
import { ledgerSaga } from 'app/state/ledger/saga'
import { stakingSaga } from 'app/state/staking/saga'
import { transactionSaga } from 'app/state/transaction/saga'
import { rootWalletSaga } from 'app/state/wallet/saga'

export default function* rootSagas() {
  yield all([accountSaga(), networkSaga(), ledgerSaga(), stakingSaga(), transactionSaga(), rootWalletSaga()])
}
