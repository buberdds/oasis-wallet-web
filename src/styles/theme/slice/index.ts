import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { useInjectReducer } from 'utils/redux-injectors'

import { getThemeFromStorage, saveTheme } from '../utils'
import { ThemeState } from './types'

export const initialState: ThemeState = {
  selected: getThemeFromStorage() || 'system',
}

const slice = createSlice({
  name: 'theme',
  initialState,
  reducers: {
    changeTheme(state, action: PayloadAction<'dark' | 'light'>) {
      saveTheme(action.payload)
      state.selected = action.payload
    },
  },
})

// export const { actions: themeActions, reducer } = slice

// export const useThemeSlice = () => {
//   useInjectReducer({ key: slice.name, reducer: slice.reducer })
//   return { actions: slice.actions }
// }
console.log(slice)
export const actions = slice.actions
export default slice.reducer
