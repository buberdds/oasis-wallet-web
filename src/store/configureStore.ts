import { configureStore } from '@reduxjs/toolkit'
import createSagaMiddleware from 'redux-saga'

import { createReducer } from './reducers'
import rootSagas from './sagas'
import { RootState } from 'types'
import { fatalErrorActions } from 'app/state/fatalerror'

export function configureAppStore(state?: Partial<RootState>) {
  const sagaMiddleware = createSagaMiddleware({
    onError: (error, info) => {
      store.dispatch(
        fatalErrorActions.setError({
          message: error.toString(),
          stack: error.stack,
          sagaStack: info.sagaStack,
        }),
      )
    },
  })

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware]

  const store = configureStore({
    reducer: createReducer(),
    middleware: getDefaultMiddleware => getDefaultMiddleware().concat(middlewares),
    devTools:
      /* istanbul ignore next line */
      process.env.NODE_ENV !== 'production',
    preloadedState: state,
  })

  sagaMiddleware.run(rootSagas)

  return store
}
