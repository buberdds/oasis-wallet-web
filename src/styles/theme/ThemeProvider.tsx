import { grommet, Grommet, ThemeType } from 'grommet'
import { deepMerge } from 'grommet/utils'
import * as React from 'react'
import { createTheme, IDataTableStyles, ITheme } from 'react-data-table-component'
import { useSelector } from 'react-redux'

import { selectTheme } from './slice/selectors'

/**
 * React-data-table by default sets its own background and text colors
 * we make sure that they do not override grommet's
 */
createTheme('blank', {
  background: { default: 'false' },
  highlightOnHover: {
    default: '#88888833',
    text: 'false',
  },
  text: {
    primary: 'false',
    secondary: 'false',
    disabled: 'false',
  },
  sortFocus: {
    default: 'false',
  },
  divider: { default: '#AAAAAAaa' },
} as Partial<ITheme>)

export const dataTableStyles: IDataTableStyles = {
  headCells: {
    style: {
      fontSize: '18px',
    },
  },
  rows: {
    style: {
      fontSize: '16x',
    },
  },
}

const customTheme: ThemeType = {
  button: {
    primary: {
      background: {
        dark: 'red',
        light: 'green',
      },
    },
    // default: {
    //   background: {
    //     dark: '#7D4CDB',
    //     light: '#7D4CDB',
    //   },
    // },
  },
  global: {
    colors: {
      oasisIndigo: '#310081',
      oasisMinty: '#4CD4A9',
      oasisLightGray: '#ececec',
      brand: {
        light: '#0092f6',
        dark: '#0092f6bb',
      },
      oasisBlue2: '#4db3f9',
      oasisBlue3: '#26a2f8',
      'status-ok': '#4cd4a9',
      'status-error': '#ff4212',
      'status-warning': '#f4ce4b',
      'background-oasis-blue': {
        dark: '#0f477b',
        light: '#0092f6',
      },
      neutral: {
        dark: '#310081FF',
        light: '#310081FF',
      },
      'neutral-2': {
        dark: '#0092f6bb',
        light: '#0092f6bb',
      },
      'background-back': {
        dark: '#1A1A2e',
        light: '#EFEFEF',
      },
      'background-front': {
        dark: '#16213e',
        light: '#FFFFFF',
      },
      'background-front-lighter': {
        dark: '#222938',
        light: '#FFFFFF',
      },
      'background-front-border': {
        dark: '#111111',
        light: '#EDEDED',
      },
      'background-contrast': {
        dark: '#FFFFFF08',
        light: '#11111108',
      },
      'background-contrast-2': {
        dark: '#FFFFFF33',
        light: '#11111133',
      },
      'background-custom': {
        dark: '#0E5265',
        light: '#00C8FF',
      },
      'component-toolbar': {
        dark: '#0f346088',
        light: '#26a2f844',
      },
      'component-sidebar': {
        dark: '#16213e',
        light: '#fafafa',
      },
    },
    font: {
      family: 'Rubik',
    },
    opacity: {
      weak: 0.25,
    },
  },
}
export const ThemeProvider = (props: { children: React.ReactChild }) => {
  const theme = deepMerge(grommet, customTheme)
  const mode = useSelector(selectTheme)

  return (
    <Grommet theme={theme} themeMode={mode} style={{ minHeight: '100vh' }}>
      {React.Children.only(props.children)}
    </Grommet>
  )
}
