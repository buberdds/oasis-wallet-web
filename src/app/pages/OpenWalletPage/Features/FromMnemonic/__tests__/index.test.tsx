import * as React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { configureAppStore } from 'store/configureStore'
import { Provider } from 'react-redux'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { FromMnemonic } from '..'

const renderPage = (store: any) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <FromMnemonic />
      </ThemeProvider>
    </Provider>,
  )

describe('<FromMnemonic/>', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
  })

  // @todo this has a random mnemonic...
  it('should match snapshot', () => {
    const page = renderPage(store)
    expect(page.container.firstChild).toMatchSnapshot()
  })

  it('should display an error on invalid mnemonic', () => {
    renderPage(store)
    const textbox = screen.getByRole('textbox') as HTMLInputElement
    const button = screen.getByRole('button')
    userEvent.type(textbox, 'hello')
    userEvent.click(button)
    const errorElem = screen.getByText('openWallet.mnemonic.error')
    expect(errorElem).toBeInTheDocument()

    // A valid phrase should remove the error
    textbox.setSelectionRange(0, 5)
    userEvent.type(textbox, 'echo toward hold roast rather reduce cute civil equal whale wait conduct')
    userEvent.click(button)
    expect(errorElem).not.toBeInTheDocument()
  })

  it('newlines in mnemonic should be valid', () => {
    renderPage(store)
    const textbox = screen.getByRole('textbox')
    const button = screen.getByRole('button')
    userEvent.type(textbox, 'echo\ntoward hold   roast\n rather reduce cute civil equal whale wait conduct')
    userEvent.click(button)
    const errorElem = screen.queryByText('openWallet.mnemonic.error')
    expect(errorElem).toBeNull()
  })
})
