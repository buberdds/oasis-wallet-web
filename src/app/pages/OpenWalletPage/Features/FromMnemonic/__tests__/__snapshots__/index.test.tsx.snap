// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FromMnemonic/> should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 12px;
  background-color: #FFFFFF;
  color: #444444;
  border: solid 1px #EDEDED;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
  border-radius: 5px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 12px;
  margin-bottom: 12px;
  margin-left: 12px;
  margin-right: 48px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background: #11111108;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 6px;
  background: #11111108;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 6px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-right: 12px;
}

.c16 {
  font-size: 18px;
  line-height: 24px;
}

.c11 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 4px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #0092f6;
  color: #f8f8f8;
  border-radius: 4px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c11:hover {
  box-shadow: 0px 0px 0px 2px #0092f6;
}

.c11:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c11:focus > circle,
.c11:focus > ellipse,
.c11:focus > line,
.c11:focus > path,
.c11:focus > polygon,
.c11:focus > polyline,
.c11:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c11:focus::-moz-focus-inner {
  border: 0;
}

.c11:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c11:focus:not(:focus-visible) > circle,
.c11:focus:not(:focus-visible) > ellipse,
.c11:focus:not(:focus-visible) > line,
.c11:focus:not(:focus-visible) > path,
.c11:focus:not(:focus-visible) > polygon,
.c11:focus:not(:focus-visible) > polyline,
.c11:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c11:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c4 {
  margin: 0;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c13 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(14ch,1fr) );
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(100%,1fr) );
  grid-gap: 12px 12px;
  padding: 12px;
}

.c0 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #444444;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c5 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
}

.c9 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  font-size: 18px;
  line-height: 24px;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  height: 100%;
}

.c9:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus > circle,
.c9:focus > ellipse,
.c9:focus > line,
.c9:focus > path,
.c9:focus > polygon,
.c9:focus > polyline,
.c9:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus::-moz-focus-inner {
  border: 0;
}

.c9::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c9::-moz-placeholder {
  color: #AAAAAA;
}

.c9:-ms-input-placeholder {
  color: #AAAAAA;
}

.c9::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c9::-moz-focus-inner {
  border: none;
  outline: none;
}

.c9:-moz-placeholder,
.c9::-moz-placeholder {
  opacity: 1;
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    border: solid 1px #EDEDED;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin-top: 6px;
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin-left: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin-right: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c10 {
    margin-top: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c14 {
    margin: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c14 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c14 {
    padding: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 6px;
  }
}

<div
  class="c0"
  style="min-height: 100vh;"
>
  <div
    class="mocked-framer-motion-div"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <form>
            <h1
              class="c4"
            >
              openWallet.mnemonic.header
            </h1>
            <p
              class="c5"
            >
              openWallet.mnemonic.instruction
            </p>
            <div
              class="c6"
            >
              <div
                class="c7 "
              >
                <div
                  class="c8 FormField__FormFieldContentBox-sc-m9hood-1"
                >
                  <div
                    class="c6"
                  >
                    <textarea
                      class="c9"
                      data-testid="mnemonic"
                      id="mnemonic"
                      placeholder="openWallet.mnemonic.enterPhraseHere"
                      rows="5"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="c10"
            >
              <button
                class="c11"
                type="submit"
              >
                openWallet.mnemonic.import
              </button>
            </div>
          </form>
        </div>
        <div
          class="c12"
        >
          <span
            class="notranslate"
            translate="no"
          >
            <div
              class="c13"
              data-testid="mnemonic-grid"
            >
              <div
                class="c14"
              >
                <div
                  class="c15"
                >
                  <span
                    class="c16"
                    style="user-select: none;"
                  >
                    1
                    .
                  </span>
                </div>
                <div
                  class="c6"
                >
                  <span
                    class="notranslate"
                    translate="no"
                  >
                    <strong
                      style="white-space: pre;"
                    />
                  </span>
                </div>
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
